pkg/reconciler/cluster/syncer.go

```diff
@@ -76,7 +137,7 @@ func installSyncer(ctx context.Context, client kubernetes.Interface, syncerImage
 
 	args := []string{
 		"-cluster", clusterID,
-		"-kubeconfig", "/kcp/kubeconfig",
+		"-from", "/kcp/kubeconfig",
```

Comment:
Can we pass the `to` kubeconfig here too? We'd just need to write its value to the ConfigMap too (from the Cluster's `.spec.kubeconfig`), and it'd let us not rely on `InClusterConfig` in real code paths.

Reply:
Not sure I agree here. The cluster `.spec.kubeconfig` probably contains the kubeconfig for "external access" to your cluster.
When some code is running inside a cluster POD, it seems more robust to me to use the InCluster config : it uses internal Kubernetes service, the already injected POD service account, etc, etc... that will avoid a number of potential problems with external access, like proxies, etc ...

Reply:
That makes sense, I'm convinced.

In that case, when we move syncer out of the target cluster and into cluster-controller, will we even need these flags? The cluster-controller can just start a syncer controller with `syncer.Controller{queue, fromDSIF, toClient}` constructed from kubeconfigs, and call `go StartWorker()` on it.

Reply:
Well, I somehow think that being able to start the syncer as a standalone command line, and *not as part of the cluster controller* is useful.
That means that is could be changed / adapted by users and started locally, without impacting the cluster-manager, which still have to manage *api resource definitions*.